crowbar_framework/app/models/service_object.rb

```diff
@@ -196,6 +196,57 @@ def restore_to_ready(nodes)
     end
   end
 
+  def reset_proposal(inst, bc = @bc_name)
+    ::Proposal.find_by(
+      barclamp: bc,
+      name: inst
+    ).tap do |proposal|
+      if proposal.nil?
+        return [
+          404,
+          I18n.t("model.service.cannot_find")
+        ]
+      end
+
+      proposal["deployment"][bc]["crowbar-committing"] = false
+
+      unless proposal.save
+        return [
+          422,
+          I18n.t("proposal.failures.proposal_reset")
+        ]
+      end
+
+      nodes = NodeObject.find("roles:#{bc}-config-#{inst}").map do |node|
+        if node.role.nil?
+          false
+        else
+          node.role.tap do |role|
+            role.default_attributes["state"] = "ready"
+            role.save
+          end
+        end
+      end
+
+      unless nodes.all?
+        return [
+          422,
+          I18n.t("proposal.failures.nodes_reset")
+        ]
+      end
+    end
+
+    [
+      200,
+      ""
+    ]
+  # rescue => e
+  #   [
+  #     500,
+  #     e.message
+  #   ]
```

Comment:
Style/CommentIndentation: Incorrect indentation detected (column 2 instead of 4).